core/tests/protractor_desktop/topicAndStoryViewer.js

```diff
@@ -0,0 +1,158 @@
+// Copyright 2018 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview End-to-end tests for the story viewer.
+ */
+
+var forms = require('../protractor_utils/forms.js');
+var general = require('../protractor_utils/general.js');
+var users = require('../protractor_utils/users.js');
+var workflow = require('../protractor_utils/workflow.js');
+
+var AdminPage = require('../protractor_utils/AdminPage.js');
+var Constants = require('../protractor_utils/ProtractorConstants.js');
+var TopicsAndSkillsDashboardPage =
+  require('../protractor_utils/TopicsAndSkillsDashboardPage.js');
+var TopicAndStoryViewerPage = require(
+  '../protractor_utils/TopicAndStoryViewerPage.js');
+var TopicEditorPage = require('../protractor_utils/TopicEditorPage.js');
+var StoryEditorPage = require('../protractor_utils/StoryEditorPage.js');
+var ExplorationPlayerPage =
+  require('../protractor_utils/ExplorationPlayerPage.js');
+
+describe('Story viewer functionality', function() {
+  var adminPage = null;
+  var topicAndStoryViewerPage = null;
+  var topicsAndSkillsDashboardPage = null;
+  var topicEditorPage = null;
+  var storyEditorPage = null;
+  var explorationPlayerPage = null;
+  var dummyExplorationIds = [];
+
+  beforeAll(async function() {
+    adminPage = new AdminPage.AdminPage();
+    explorationPlayerPage = new ExplorationPlayerPage.ExplorationPlayerPage();
+    topicAndStoryViewerPage = (
+      new TopicAndStoryViewerPage.TopicAndStoryViewerPage());
+    topicsAndSkillsDashboardPage = (
+      new TopicsAndSkillsDashboardPage.TopicsAndSkillsDashboardPage());
+    topicEditorPage = new TopicEditorPage.TopicEditorPage();
+    storyEditorPage = new StoryEditorPage.StoryEditorPage();
+  });
+
+  var createDummyExplorations = async function() {
+    var EXPLORATION = {
+      category: 'Learning',
+      objective: 'The goal is to check story viewer functionality.',
+      language: 'English'
+    };
+    for (var i = 1; i <= 3; i++) {
+      await workflow.createAndPublishTwoCardExploration(
+        `Exploration SPF1 - ${i}`,
```

Comment:
What does the SPF1 mean? Also, can we use `topicAndStoryViewer` in the exploration name for uniqueness?